pkg/controllers/nodeclass/status/ami.go

```diff
@@ -42,7 +42,9 @@ func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (reconc
 	if len(amis) == 0 {
 		nodeClass.Status.AMIs = nil
 		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeAMIsReady, "AMINotFound", "AMISelector did not match any AMIs")
-		return reconcile.Result{}, nil
+		// If users have omitted the necessary tags and later add them, we need to reprocess the information.
+		// Returning 'ok' in this case means that the nodeclass will remain in an unready state until the component is restarted.
+		return reconcile.Result{RequeueAfter: 15 * time.Second}, nil
```

Comment:
Yeah, this makes sense -- definitely agree that we should be re-polling. I'm wondering if we should just use the same RequeueAfter as we do in the successful case -- since there's really no reason to poll more aggressively here.

Just generally, I think we should be doing time.Minute RequeueAfter since that's our cache timeout interval so polling at that frequency can ensure that we get refreshed data more frequently. This should apply across the reconcilers -- subnet, security group, and amis

Reply:
thanks for your reminder, i have check the cache timeout its 2 minues,